flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionVertexDeploymentOption.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler;
+
+import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
+
+/**
+ * Component that stores the task need to be scheduled and the option for deployment.
+ */
+public class ExecutionVertexDeploymentOption {
+
+	private final ExecutionVertexID executionVertexId;
+
+	private final DeploymentOption deploymentOption;
+
+	public ExecutionVertexDeploymentOption(ExecutionVertexID executionVertexId, DeploymentOption deploymentOption) {
```

Comment:
Currently we only have one kind of DeploymentOption (ExecutionVertexDeploymentOption) and the SchedulingStrategy is bind to ExecutionVertexID as well. Is it necessary to have a DeploymentOption?
B.T.W. If it is necessary, why using composition here rather than inheritance?

Reply:
Sorry I made a mistake here. I was bit confused of DeploymentOption and ExecutionVertexDeploymentOption due to the similar name. Now I understand they actually are different components. ExecutionVertexDeploymentOption is a deployment descriptor and the DeploymentOption is a wrapper of the options.

Just wonder whether we can have different name patterns for them, like ExecutionVertexDeploymentDescriptor and DeploymentOption?

Reply:
Ok, I can see that the naming can cause confusion.  `ExecutionVertexDeploymentOption` is meant to be understood as a _pair_ of an `ExecutionVertex` and a `DeploymentOption`. The name you suggested (`ExecutionVertexDeploymentDescriptor`) works for me.

Reply:
What do you think @shuai-xu?

Reply:
I think both name works, we can keep it as now.